vulnerabilities/scraper/safety_db.py

```diff
@@ -0,0 +1,60 @@
+# Data Imported from https://github.com/pyupio/safety-db
+import json
+from urllib.error import HTTPError
+from urllib.request import urlopen
+from dephell_specifier import RangeSpecifier
+
+URL = 'https://raw.githubusercontent.com/pyupio/safety-db/master/data/insecure_full.json'
+
+
+def get_data_file():
+    with urlopen(URL) as file:
+        json_file = json.load(file)
+    return json_file
+
+
+def get_all_versions_of_package(package_name):
+    url = "https://pypi.org/pypi/{}/json".format(package_name)
+    try:
+        with urlopen(url) as response:
+            json_file = json.load(response)
+    except HTTPError:  # PyPi does not have data about this package, we skip these
+        return set()
+    for release in json_file['releases']:
+        yield release
+
+
+def import_vulnerabilities():
+    vulnerability_package_dicts = []
+    data = get_data_file()
+    cves = set()
+    for package in data:
+        package_name = package
+        all_package_versions = set(get_all_versions_of_package(package_name))
+        if len(all_package_versions) == 0:
+            # PyPi does not have data about this package, we skip these
+            continue
+        for advisory in data[package]:
+            description = advisory['advisory']
+            cve_id = advisory['cve'] if advisory['cve'] else None
+            vuln_id = advisory['id']
+            vuln_version_ranges = advisory['specs']
+            for vuln_version_range in vuln_version_ranges:
+                version_range = RangeSpecifier(vuln_version_range)
+                affected_versions = set()
+                for version in all_package_versions:
+                    if version in version_range:
+                        affected_versions.add(version)
+            unaffected_versions = all_package_versions - affected_versions
+            if cve_id not in cves:
```

Comment:
If two packages are affected by the same CVE, doesn't this check cause the second one to be skipped entirely? And didn't this issue come up before somewhere? It seems familiar...

Reply:
You are correct, I remember vaguely, that the code was violating constraints which was temporarily fixed by removing the `summary`. We need some concrete way to deal with this. Some pointers to duplicate(not exactly) data https://github.com/pyupio/safety-db/blob/b873cceee96c04c4e9df93d17087592e94290482/data/insecure_full.json#L247  and   https://github.com/pyupio/safety-db/blob/b873cceee96c04c4e9df93d17087592e94290482/data/insecure_full.json#L256

Reply:
Let's merge it as is and refine it later.